### ISSUE TITLE
Error during startup

### ISSUE THREAD

**Issue by BratishkaErik** [LielAmar/2FA#49]
```
[23:22:37 ERROR]: Error occurred while enabling 2FA v1.4.2 (Is it up to date?)
java.lang.NoSuchMethodError: com.zaxxer.hikari.HikariDataSource.setKeepaliveTime(J)V
        at com.lielamar.auth.shared.storage.sql.SQLStorage.setupHikari(SQLStorage.java:60) ~[?:?]
        at com.lielamar.auth.shared.storage.sql.SQLStorage.<init>(SQLStorage.java:48) ~[?:?]
        at com.lielamar.auth.shared.storage.StorageHandler.loadStorageHandler(StorageHandler.java:93) ~[?:?]
        at com.lielamar.auth.bukkit.TwoFactorAuthentication.setupAuth(TwoFactorAuthentication.java:64) ~[?:?]
        at com.lielamar.auth.bukkit.TwoFactorAuthentication.onEnable(TwoFactorAuthentication.java:39) ~[?:?]
        at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:264) ~[patched_1.12.2.jar:git-Paper-1618]
        at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:316) ~[patched_1.12.2.jar:git-Paper-1618]
        at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:405) ~[patched_1.12.2.jar:git-Paper-1618]
        at org.bukkit.craftbukkit.v1_12_R1.CraftServer.enablePlugin(CraftServer.java:395) ~[patched_1.12.2.jar:git-Paper-1618]
        at org.bukkit.craftbukkit.v1_12_R1.CraftServer.enablePlugins(CraftServer.java:344) ~[patched_1.12.2.jar:git-Paper-1618]
        at net.minecraft.server.v1_12_R1.MinecraftServer.t(MinecraftServer.java:442) ~[patched_1.12.2.jar:git-Paper-1618]
        at net.minecraft.server.v1_12_R1.MinecraftServer.l(MinecraftServer.java:403) ~[patched_1.12.2.jar:git-Paper-1618]
        at net.minecraft.server.v1_12_R1.MinecraftServer.a(MinecraftServer.java:341) ~[patched_1.12.2.jar:git-Paper-1618]
        at net.minecraft.server.v1_12_R1.DedicatedServer.init(DedicatedServer.java:289) ~[patched_1.12.2.jar:git-Paper-1618]
        at net.minecraft.server.v1_12_R1.MinecraftServer.run(MinecraftServer.java:616) ~[patched_1.12.2.jar:git-Paper-1618]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_292]
```

Happens on both lobby and arcade servers, but on the lobby server the 2fa command works, and on the arcade warns:

```
[23:22:48 WARN]: Unexpected exception while parsing console command "2fa"
org.bukkit.command.CommandException: Cannot execute command '2fa' in plugin 2FA v1.4.2 - plugin is disabled.
        at org.bukkit.command.PluginCommand.execute(PluginCommand.java:36) ~[patched_1.12.2.jar:git-Paper-1618]
        at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:152) ~[patched_1.12.2.jar:git-Paper-1618]
        at org.bukkit.craftbukkit.v1_12_R1.CraftServer.dispatchCommand(CraftServer.java:685) ~[patched_1.12.2.jar:git-Paper-1618]
        at org.bukkit.craftbukkit.v1_12_R1.CraftServer.dispatchServerCommand(CraftServer.java:648) ~[patched_1.12.2.jar:git-Paper-1618]
        at net.minecraft.server.v1_12_R1.DedicatedServer.aP(DedicatedServer.java:463) ~[patched_1.12.2.jar:git-Paper-1618]
        at net.minecraft.server.v1_12_R1.DedicatedServer.D(DedicatedServer.java:424) ~[patched_1.12.2.jar:git-Paper-1618]
        at net.minecraft.server.v1_12_R1.MinecraftServer.C(MinecraftServer.java:774) ~[patched_1.12.2.jar:git-Paper-1618]
        at net.minecraft.server.v1_12_R1.MinecraftServer.run(MinecraftServer.java:666) ~[patched_1.12.2.jar:git-Paper-1618]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_292]
```

My config:
https://hastebin.com/bedujareqa.yaml

**Comment by BratishkaErik** [comment]
Maybe I need to delete "lib/" directory?

**Comment by LielAmar** [comment]
Any chance you're using Java 16 with versions before 1.16.5?

**Comment by BratishkaErik** [comment]
> Any chance you're using Java 16 with versions before 1.16.5?

No, I'm definiteli using Java 8 with PaperSpigot 1.12.2

**Comment by LielAmar** [comment]
Alright. Deleting the /libs/ folder shouldn't really change anything but you can try that.
I will, however, try to replicate the error and see if I can find anything.

If you get any other info you think might be useful let me know! :)

**Comment by BratishkaErik** [comment]
> Alright. Deleting the /libs/ folder shouldn't really change anything but you can try that.
> I will, however, try to replicate the error and see if I can find anything.
> 
> If you get any other info you think might be useful let me know! :)

My plugins:

1. Arcade — 2FA (red, not enabled), AncientGates, BetterChairs, BKCommonLib, BungeeGuard, Chatty, Citizens, CraftEnhance, Essentials, EssentialsSpawn, IPWhitelist, IRandomTeleport, LuckPerms, Maplands, MineResetLite, MoneyNote, NametagEdit, Panilla, PlaceholderAPI, ProtocolLib, ReActions, SkinsRestorer, SmoothSleep, Vault, WorldEdit, WorldGuard
2. Lobby — 2FA (green, enabled), AncientGates, BungeeGuard, HolographicDisplays, HubBasics, IPWhitelist, LuckPerms, Panilla, PlaceholderAPI, SkinsRestore
3. Waterfall proxy (BungeeCord fork) — 2FA, BungeeGuard, BungeeKick, HubBasics-Bungee, LuckPerms-Bungee, SkinsRestorer

**Comment by BratishkaErik** [comment]
> Alright. Deleting the /libs/ folder shouldn't really change anything but you can try that.
> I will, however, try to replicate the error and see if I can find anything.
> 
> If you get any other info you think might be useful let me know! :)
Now the error occurs only on the arcade server, on the lobby it has disappeared (I didn't touch anything)

**Comment by BratishkaErik** [comment]
> If you get any other info you think might be useful let me know! :)

The error **doesn't** occur on the server, if it is the only plugin! I think it's some kind of incompatibility with another plugin, now I'll look for which one

**Comment by LielAmar** [comment]
That's really helpful!
I think I might know what's causing the problem.
I will only be able to start working on it tomorrow, but shouldn't take that much time.
If you can narrow it down and find the complicating plugin it'd confirm/deny my suspension, but if it's too much of a bother it shouldn't be necessary :)

**Comment by BratishkaErik** [comment]
> That's really helpful!
> I think I might know what's causing the problem.
> I will only be able to start working on it tomorrow, but shouldn't take that much time.
> If you can narrow it down and find the complicating plugin it'd confirm/deny my suspension, but if it's too much of a bother it shouldn't be necessary :)

Yes, I found out! The reason was the LuckPerms plugin! I checked, it doesn't matter if it has H2, YAML or MariaDB storage - together they crash the server

**Comment by LielAmar** [comment]
> > That's really helpful!
> > I think I might know what's causing the problem.
> > I will only be able to start working on it tomorrow, but shouldn't take that much time.
> > If you can narrow it down and find the complicating plugin it'd confirm/deny my suspension, but if it's too much of a bother it shouldn't be necessary :)
> 
> Yes, I found out! The reason was the LuckPerms plugin! I checked, it doesn't matter if it has H2, YAML or MariaDB storage (even though I thought there was a semblance of a data base access race, I hedged and checked other storage methods) — together they crash the server

just as I thought :)
LuckPerms is using a similar way of loading dependencies, and I guess the overlap.
I can solve it as soon as I get back home :)

**Comment by BratishkaErik** [comment]
> > > That's really helpful!
> > > I think I might know what's causing the problem.
> > > I will only be able to start working on it tomorrow, but shouldn't take that much time.
> > > If you can narrow it down and find the complicating plugin it'd confirm/deny my suspension, but if it's too much of a bother it shouldn't be necessary :)
> > 
> > 
> > Yes, I found out! The reason was the LuckPerms plugin! I checked, it doesn't matter if it has H2, YAML or MariaDB storage (even though I thought there was a semblance of a data base access race, I hedged and checked other storage methods) — together they crash the server
> 
> just as I thought :)
> LuckPerms is using a similar way of loading dependencies, and I guess the overlap.
> I can solve it as soon as I get back home :)

Ok, but why this error does not appear on the lobby server, although there are also 2FA and LuckPerms, and the same configuration? Because there are fewer plugins?

**Comment by LielAmar** [comment]
> > > > That's really helpful!
> > > > I think I might know what's causing the problem.
> > > > I will only be able to start working on it tomorrow, but shouldn't take that much time.
> > > > If you can narrow it down and find the complicating plugin it'd confirm/deny my suspension, but if it's too much of a bother it shouldn't be necessary :)
> > > 
> > > 
> > > Yes, I found out! The reason was the LuckPerms plugin! I checked, it doesn't matter if it has H2, YAML or MariaDB storage (even though I thought there was a semblance of a data base access race, I hedged and checked other storage methods) — together they crash the server
> > 
> > just as I thought :)
> > LuckPerms is using a similar way of loading dependencies, and I guess the overlap.
> > I can solve it as soon as I get back home :)
> 
> Ok, but why this error does not appear on the lobby server, although there are also 2FA and LuckPerms, and the same configuration? Because there are fewer plugins?

I have no idea 😅
maybe there's another plugin conflicting but I can put my finger on anything specific.

**Comment by BratishkaErik** [comment]
> > > > That's really helpful!
> > > > I think I might know what's causing the problem.
> > > > I will only be able to start working on it tomorrow, but shouldn't take that much time.
> > > > If you can narrow it down and find the complicating plugin it'd confirm/deny my suspension, but if it's too much of a bother it shouldn't be necessary :)
> > > 
> > > 
> > > Yes, I found out! The reason was the LuckPerms plugin! I checked, it doesn't matter if it has H2, YAML or MariaDB storage (even though I thought there was a semblance of a data base access race, I hedged and checked other storage methods) — together they crash the server
> > 
> > 
> > just as I thought :)
> > LuckPerms is using a similar way of loading dependencies, and I guess the overlap.
> > I can solve it as soon as I get back home :)
> 
> Ok, but why this error does not appear on the lobby server, although there are also 2FA and LuckPerms, and the same configuration? Because there are fewer plugins?

I reduced the number of plugins on problematic server from 26 to 23, and the problem disappeared... I removed the longest loading plugins — EssentialsX, then I removed EssentialsXChat, MoneyNote, BKCommonLib and MapLands...

UPD: Now I have made a server ONLY with these plugins, and the problem doesn't appear - judging from this, the problem is only related to the number of plugins AND LuckPerms

**Comment by LielAmar** [comment]
Hey, would you be able to join my discord server so we can continue our conversation there, and test some potential fixes? :)

**Comment by BratishkaErik** [comment]
Ok

1 августа 2021 г. 15:22:15 GMT+06:00, Liel Amar ***@***.***> пишет:

**Comment by fhebuterne** [comment]
Hi,

Same issue here with Java 8 and Paper 1.8.8 

> [00:37:15] [Server thread/ERROR]: Error occurred while enabling 2FA v1.5.1 (Is it up to date?)
java.lang.NoClassDefFoundError: org/slf4j/LoggerFactory
	at com.zaxxer.hikari.HikariConfig.<clinit>(HikariConfig.java:51) ~[?:?]
	at com.lielamar.auth.shared.storage.sql.SQLStorage.setupHikari(SQLStorage.java:57) ~[?:?]
	at com.lielamar.auth.shared.storage.sql.SQLStorage.<init>(SQLStorage.java:49) ~[?:?]
	at com.lielamar.auth.shared.storage.StorageHandler.loadStorageHandler(StorageHandler.java:91) ~[?:?]
	at com.lielamar.auth.bukkit.TwoFactorAuthentication.setupAuth(TwoFactorAuthentication.java:68) ~[?:?]
	at com.lielamar.auth.bukkit.TwoFactorAuthentication.onEnable(TwoFactorAuthentication.java:37) ~[?:?]
	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:321) ~[patched.jar:git-PaperSpigot-"4c7641d"]
	at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:332) [patched.jar:git-PaperSpigot-"4c7641d"]
	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:407) [patched.jar:git-PaperSpigot-"4c7641d"]
	at org.bukkit.craftbukkit.v1_8_R3.CraftServer.loadPlugin(CraftServer.java:359) [patched.jar:git-PaperSpigot-"4c7641d"]
	at org.bukkit.craftbukkit.v1_8_R3.CraftServer.enablePlugins(CraftServer.java:318) [patched.jar:git-PaperSpigot-"4c7641d"]
	at net.minecraft.server.v1_8_R3.MinecraftServer.s(MinecraftServer.java:408) [patched.jar:git-PaperSpigot-"4c7641d"]
	at net.minecraft.server.v1_8_R3.MinecraftServer.k(MinecraftServer.java:372) [patched.jar:git-PaperSpigot-"4c7641d"]
	at net.minecraft.server.v1_8_R3.MinecraftServer.a(MinecraftServer.java:327) [patched.jar:git-PaperSpigot-"4c7641d"]
	at net.minecraft.server.v1_8_R3.DedicatedServer.init(DedicatedServer.java:267) [patched.jar:git-PaperSpigot-"4c7641d"]
	at net.minecraft.server.v1_8_R3.MinecraftServer.run(MinecraftServer.java:563) [patched.jar:git-PaperSpigot-"4c7641d"]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_272]
Caused by: java.lang.ClassNotFoundException: org.slf4j.LoggerFactory
	at java.net.URLClassLoader.findClass(URLClassLoader.java:382) ~[?:1.8.0_272]
	at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:102) ~[patched.jar:git-PaperSpigot-"4c7641d"]
	at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:87) ~[patched.jar:git-PaperSpigot-"4c7641d"]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:418) ~[?:1.8.0_272]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:351) ~[?:1.8.0_272]
	... 17 more

And same issue but not same dependencies with Java 11 et Paper 1.16.5 

> [00:44:25] [Server thread/ERROR]: Error occurred while enabling 2FA v1.5.1 (Is it up to date?)
> java.lang.NoClassDefFoundError: com/zaxxer/hikari/HikariDataSource
> 	at com.lielamar.auth.shared.storage.sql.SQLStorage.setupHikari(SQLStorage.java:57) ~[?:?]
> 	at com.lielamar.auth.shared.storage.sql.SQLStorage.<init>(SQLStorage.java:49) ~[?:?]
> 	at com.lielamar.auth.shared.storage.StorageHandler.loadStorageHandler(StorageHandler.java:91) ~[?:?]
> 	at com.lielamar.auth.bukkit.TwoFactorAuthentication.setupAuth(TwoFactorAuthentication.java:68) ~[?:?]
> 	at com.lielamar.auth.bukkit.TwoFactorAuthentication.onEnable(TwoFactorAuthentication.java:37) ~[?:?]
> 	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:263) ~[patched_1.16.5.jar:git-Paper-573]
> 	at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:380) ~[patched_1.16.5.jar:git-Paper-573]
> 	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:483) ~[patched_1.16.5.jar:git-Paper-573]
> 	at org.bukkit.craftbukkit.v1_16_R3.CraftServer.enablePlugin(CraftServer.java:501) ~[patched_1.16.5.jar:git-Paper-573]
> 	at org.bukkit.craftbukkit.v1_16_R3.CraftServer.enablePlugins(CraftServer.java:415) ~[patched_1.16.5.jar:git-Paper-573]
> 	at net.minecraft.server.v1_16_R3.MinecraftServer.loadWorld(MinecraftServer.java:580) ~[patched_1.16.5.jar:git-Paper-573]
> 	at net.minecraft.server.v1_16_R3.DedicatedServer.init(DedicatedServer.java:280) ~[patched_1.16.5.jar:git-Paper-573]
> 	at net.minecraft.server.v1_16_R3.MinecraftServer.w(MinecraftServer.java:1055) ~[patched_1.16.5.jar:git-Paper-573]
> 	at net.minecraft.server.v1_16_R3.MinecraftServer.lambda$a$0(MinecraftServer.java:289) ~[patched_1.16.5.jar:git-Paper-573]
> 	at java.lang.Thread.run(Thread.java:829) [?:?]
> Caused by: java.lang.ClassNotFoundException: com.zaxxer.hikari.HikariDataSource
> 	at java.net.URLClassLoader.findClass(URLClassLoader.java:471) ~[?:?]
> 	at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:182) ~[patched_1.16.5.jar:git-Paper-573]
> 	at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:111) ~[patched_1.16.5.jar:git-Paper-573]
> 	at java.lang.ClassLoader.loadClass(ClassLoader.java:589) ~[?:?]
> 	at java.lang.ClassLoader.loadClass(ClassLoader.java:522) ~[?:?]
> 	... 15 more

**Comment by LielAmar** [comment]
> Hi,
> 
> Same issue here with Java 8 and Paper 1.8.8 
> 
> > [00:37:15] [Server thread/ERROR]: Error occurred while enabling 2FA v1.5.1 (Is it up to date?)
> java.lang.NoClassDefFoundError: org/slf4j/LoggerFactory
> 	at com.zaxxer.hikari.HikariConfig.<clinit>(HikariConfig.java:51) ~[?:?]
> 	at com.lielamar.auth.shared.storage.sql.SQLStorage.setupHikari(SQLStorage.java:57) ~[?:?]
> 	at com.lielamar.auth.shared.storage.sql.SQLStorage.<init>(SQLStorage.java:49) ~[?:?]
> 	at com.lielamar.auth.shared.storage.StorageHandler.loadStorageHandler(StorageHandler.java:91) ~[?:?]
> 	at com.lielamar.auth.bukkit.TwoFactorAuthentication.setupAuth(TwoFactorAuthentication.java:68) ~[?:?]
> 	at com.lielamar.auth.bukkit.TwoFactorAuthentication.onEnable(TwoFactorAuthentication.java:37) ~[?:?]
> 	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:321) ~[patched.jar:git-PaperSpigot-"4c7641d"]
> 	at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:332) [patched.jar:git-PaperSpigot-"4c7641d"]
> 	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:407) [patched.jar:git-PaperSpigot-"4c7641d"]
> 	at org.bukkit.craftbukkit.v1_8_R3.CraftServer.loadPlugin(CraftServer.java:359) [patched.jar:git-PaperSpigot-"4c7641d"]
> 	at org.bukkit.craftbukkit.v1_8_R3.CraftServer.enablePlugins(CraftServer.java:318) [patched.jar:git-PaperSpigot-"4c7641d"]
> 	at net.minecraft.server.v1_8_R3.MinecraftServer.s(MinecraftServer.java:408) [patched.jar:git-PaperSpigot-"4c7641d"]
> 	at net.minecraft.server.v1_8_R3.MinecraftServer.k(MinecraftServer.java:372) [patched.jar:git-PaperSpigot-"4c7641d"]
> 	at net.minecraft.server.v1_8_R3.MinecraftServer.a(MinecraftServer.java:327) [patched.jar:git-PaperSpigot-"4c7641d"]
> 	at net.minecraft.server.v1_8_R3.DedicatedServer.init(DedicatedServer.java:267) [patched.jar:git-PaperSpigot-"4c7641d"]
> 	at net.minecraft.server.v1_8_R3.MinecraftServer.run(MinecraftServer.java:563) [patched.jar:git-PaperSpigot-"4c7641d"]
> 	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_272]
> Caused by: java.lang.ClassNotFoundException: org.slf4j.LoggerFactory
> 	at java.net.URLClassLoader.findClass(URLClassLoader.java:382) ~[?:1.8.0_272]
> 	at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:102) ~[patched.jar:git-PaperSpigot-"4c7641d"]
> 	at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:87) ~[patched.jar:git-PaperSpigot-"4c7641d"]
> 	at java.lang.ClassLoader.loadClass(ClassLoader.java:418) ~[?:1.8.0_272]
> 	at java.lang.ClassLoader.loadClass(ClassLoader.java:351) ~[?:1.8.0_272]
> 	... 17 more
> 
> And same issue on other server but not same dependencies with Java 11 et Paper 1.16.5 
> 
> > [00:44:25] [Server thread/ERROR]: Error occurred while enabling 2FA v1.5.1 (Is it up to date?)
> > java.lang.NoClassDefFoundError: com/zaxxer/hikari/HikariDataSource
> > 	at com.lielamar.auth.shared.storage.sql.SQLStorage.setupHikari(SQLStorage.java:57) ~[?:?]
> > 	at com.lielamar.auth.shared.storage.sql.SQLStorage.<init>(SQLStorage.java:49) ~[?:?]
> > 	at com.lielamar.auth.shared.storage.StorageHandler.loadStorageHandler(StorageHandler.java:91) ~[?:?]
> > 	at com.lielamar.auth.bukkit.TwoFactorAuthentication.setupAuth(TwoFactorAuthentication.java:68) ~[?:?]
> > 	at com.lielamar.auth.bukkit.TwoFactorAuthentication.onEnable(TwoFactorAuthentication.java:37) ~[?:?]
> > 	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:263) ~[patched_1.16.5.jar:git-Paper-573]
> > 	at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:380) ~[patched_1.16.5.jar:git-Paper-573]
> > 	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:483) ~[patched_1.16.5.jar:git-Paper-573]
> > 	at org.bukkit.craftbukkit.v1_16_R3.CraftServer.enablePlugin(CraftServer.java:501) ~[patched_1.16.5.jar:git-Paper-573]
> > 	at org.bukkit.craftbukkit.v1_16_R3.CraftServer.enablePlugins(CraftServer.java:415) ~[patched_1.16.5.jar:git-Paper-573]
> > 	at net.minecraft.server.v1_16_R3.MinecraftServer.loadWorld(MinecraftServer.java:580) ~[patched_1.16.5.jar:git-Paper-573]
> > 	at net.minecraft.server.v1_16_R3.DedicatedServer.init(DedicatedServer.java:280) ~[patched_1.16.5.jar:git-Paper-573]
> > 	at net.minecraft.server.v1_16_R3.MinecraftServer.w(MinecraftServer.java:1055) ~[patched_1.16.5.jar:git-Paper-573]
> > 	at net.minecraft.server.v1_16_R3.MinecraftServer.lambda$a$0(MinecraftServer.java:289) ~[patched_1.16.5.jar:git-Paper-573]
> > 	at java.lang.Thread.run(Thread.java:829) [?:?]
> > Caused by: java.lang.ClassNotFoundException: com.zaxxer.hikari.HikariDataSource
> > 	at java.net.URLClassLoader.findClass(URLClassLoader.java:471) ~[?:?]
> > 	at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:182) ~[patched_1.16.5.jar:git-Paper-573]
> > 	at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:111) ~[patched_1.16.5.jar:git-Paper-573]
> > 	at java.lang.ClassLoader.loadClass(ClassLoader.java:589) ~[?:?]
> > 	at java.lang.ClassLoader.loadClass(ClassLoader.java:522) ~[?:?]
> > 	... 15 more
> 

What version of the plugin are you using? Can you provide the full log with everything that's been sent regarding 2FA?

**Comment by LielAmar** [comment]
All known problems in this issue were fixed on version 1.5.2 :)
